helpers/uboot.sh

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2018 Evgeny Golyshev <eugulixes@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Checks if the U-Boot dependencies are satisfied.
+# Globals:
+#     None
+# Arguments:
+#     None
+# Returns:
+#     0 or 1 depending on the result.
+are_uboot_dependencies_satisfied() {
```

Comment:
The function lacks checking of
* the `python` executable;
* the `swig` executable;
* the `/usr/include/python2.7/Python.h` header (available in `python-dev` on Debian/Ubuntu and in `python2-devel` on Fedora).

You may check the need of the dependencies trying to build U-Boot a clean environment (for example, `docker run -it --run debian:stretch bash`).